src/Abp.Zero.EntityFramework/EntityHistory/EntityHistoryHelper.cs

```diff
@@ -487,54 +475,94 @@ private void UpdateChangeSet(DbContext context, EntityChangeSet changeSet)
+                    {
+                        var newPropertyChange = CreateEntityPropertyChange(oldValue, newValue, propertyInfo);
+                        if (propertyChange.NewValue == newPropertyChange.NewValue)
+                        {
+                            // No change
+                            entityChange.PropertyChanges.Remove(propertyChange);
+                        }
+                        else
                         {
-                            if (propertyEntry.HasChanged())
-                            {
-                                var propertyInfo = entityEntry.GetPropertyInfo(property.Name);
-
-                                // Add foreign key
-                                entityChange.PropertyChanges.Add(new EntityPropertyChange
-                                {
-                                    NewValue = propertyEntry.CurrentValue.ToJsonString(),
-                                    OriginalValue = propertyEntry.OriginalValue.ToJsonString(),
-                                    PropertyName = property.Name,
-                                    PropertyTypeFullName = propertyInfo.PropertyType.FullName
-                                });
-                            }
-
-                            continue;
+                            // Update foreign key
+                            propertyChange.NewValue = newPropertyChange.NewValue;
                         }
+                    }
+                }
+
+                /* Update tracked properties */
+                var auditedPropertyNames = entityChange.PropertyChanges
+                                             .Select(pc => pc.PropertyName)
+                                             .Except(foreignKeyPropertyNames)
+                                             .Distinct()
+                                             .ToList();
 
-                        if (propertyChange.OriginalValue == propertyChange.NewValue)
+                foreach(var propertyName in auditedPropertyNames)
+                {
+                    var memberEntry = entityEntry.Member(propertyName);
+                    var primitiveOrComplexProperty = memberEntry is DbPropertyEntry || memberEntry is DbComplexPropertyEntry;
+                    if (!primitiveOrComplexProperty)
+                    {
+                        continue;
+                    }
+
+                    var propertyEntry = entityEntry.Property(propertyName);
+                    var propertyInfo = entityEntry.GetPropertyInfo(propertyName);
+                    var propertyChange = entityChange.PropertyChanges.FirstOrDefault(pc => pc.PropertyName == propertyName);
+                    var oldValue = propertyEntry.GetOriginalValue();
+                    var newValue = propertyEntry.GetNewValue();
+
+                    if (propertyChange != null && propertyChange.OriginalValue == propertyChange.NewValue)
```

Comment:
```suggestion
                    if (propertyChange.OriginalValue == propertyChange.NewValue)
```

As mentioned above, `propertyChange` must exist.